test/unit/grammar/range_rule.cpp

```diff
@@ -10,23 +10,57 @@
 // Test that header file is self-contained.
 #include <boost/url/grammar/range_rule.hpp>
 
+#include <boost/url/grammar/alpha_chars.hpp>
 #include <boost/url/grammar/char_rule.hpp>
 #include <boost/url/grammar/parse.hpp>
+#include <boost/url/grammar/sequence_rule.hpp>
+#include <boost/url/grammar/token_rule.hpp>
 
 #include "test_suite.hpp"
 
+#include <algorithm>
+#include <initializer_list>
+
 namespace boost {
 namespace urls {
 namespace grammar {
 
 struct range_rule_test
 {
+    template<class R>
+    static
+    void
+    check(
+        string_view s,
+        std::initializer_list<
+            string_view> init,
+        R const& r)
+    {
+        auto rv = parse(s, r);
+        if(! BOOST_TEST(rv.has_value()))
+            return;
+        if(! BOOST_TEST_EQ(
+                rv->size(), init.size()))
+            return;
+        BOOST_TEST(
+            std::equal(
+                rv->begin(),
+                rv->end(),
+                init.begin()));
+    }
+
     void
     run()
     {
         // test constexpr
-        constexpr auto r =
-            range_rule(char_rule('.'));
+        constexpr auto r = range_rule(
+            token_rule(alpha_chars),
+            sequence_rule(
+                char_rule('+'),
+                token_rule(alpha_chars)));
+
+        check("", {}, r);
+        check("x", {"x"}, r);
```

Comment:
This test doesn't seem exhaustive enough to me. Unless I'm misreading, I'd consider adding further cases:
```cpp
check("a+b+c", {"a", "b", "c"}, r);
check("a+b+", {"a", "b", "c"}, r);
check("+b+c", {"a", "b", "c"}, r);
check("a++c", {"a", "b", "c"}, r);
```
or something to this effect.

Reply:
Yeah, I am furiously trying to finish implementation, design, and docs - tests are kind of an afterthought since they can wait until after the review. Truth be told, this range gets so much testing from being used for query params and path segments that I am not too worried about it. But yes we will circle back and add more.